examples/fft_physical_x/fft_grid_x.f90

```diff
@@ -47,10 +49,15 @@ program fft_physical_x
    ! output is Z-pencil data
    call alloc_x(in, ph, .true.)
    call alloc_z(out, ph, .true.)
+   ! Convert pointers to loops start/end to scalar
```

Comment:
This is mandatory for GPUs ? This modification is missing in other examples ?

Reply:
Since loop dimensions are all stored in arrays we have two options to run them on GPU: 1 convert loop start/end into scalars (the one used above) or move the array to device (used for example here) https://github.com/xcompact3d/2decomp-fft/blob/f9d047fcc998f3737cb3ce61c3e04848b12751ad/examples/test2d/test2d.f90#L48
I wanted to give example with both options but i understand that this look inconsistent. Open to suggestion to what is the best way.

Reply:
If the two solutions are equivalent, I think we have two options :

1. Use only the recommended one in the examples
2. Use both in the examples and explain somewhere there are two options

Reply:
I think it is confusing to have both options. In my opinion we should not pass to GPU very small arrays like `xstat` or `xend`. I’ll modify all the examples having only local scalars as loop boundaries

Reply:
I’m doing the examples with local variables. On Thursday we can decide a unique way forward and we can do all relative modifications.